url79d4f7c7ed_schanq_Live_Algorithm_tgz-pJ8-algorithms_and_systems_AI_Evolutionary_GeneticAlgorithmJ8/src/networking/OSCClient.java

```diff
@@ -0,0 +1,344 @@
+// ##################################################################
+// ##################################################################
+// # OSCClient: code based on oscP5 examples by andreas schlegel	#
+// # Receives incoming messages from a UDP send object in Max and	#
+// # submits the data to the fuzzy system and genetic algorithm for	#
+// # analysis and generation of a response.							#
+// ##################################################################
+// ##################################################################
+package networking;
+
+import java.util.Arrays;
+
+import algorithms_and_systems_AI_Evolutionary.Chromosome;
+import algorithms_and_systems_AI_Evolutionary.FuzzySystem;
+import algorithms_and_systems_AI_Evolutionary.GeneticAlgorithm;
+import algorithms_and_systems_AI_Evolutionary.TargetChromosome;
+import oscP5.OscMessage;
+import oscP5.OscP5;
+import netP5.NetAddress;
+
+public class OSCClient {
+
+	// ##########################################################
+	// #						FIELDS							#
+	// ##########################################################
+	public OscP5 oscP5;
+	public NetAddress myRemoteLocation;
+	
+	int minimumRange, maximumRange;
+	double BPM;
+	
+	public int [] noteValues;
+	public int [] velocityValues;
+	public double [] millisValues;
+	
+	public boolean 	areTheNotesStored = true,
+					areTheVelocitiesStored = true, 
+					areTheMillisStored = true, 
+					isTheRangeSet = true, 
+					isTheBPMSet = true, 
+					readyToGo = true;
+	
+	public FuzzySystem 		theFuzzySystem;
+	public GeneticAlgorithm theGeneticAlgorithm;
+	
+	String response;
+	
+	// ##########################################################
+	// #					  CONSTRUCTOR						#
+	// ##########################################################
+	public OSCClient() {
+
+		int port 				= 59998;
+		oscP5 					= new OscP5(this, port);
+		
+		noteValues				= new int [1];
+		velocityValues 			= new int [1];
+		millisValues 			= new double [1];
+		
+		areTheNotesStored 		= false;
+		areTheVelocitiesStored	= false;
+		areTheMillisStored		= false;
+		isTheRangeSet			= false;
+		isTheBPMSet				= false;
+		readyToGo				= false;
+		
+		theFuzzySystem = new FuzzySystem();
+		
+		response = "";
+		
+		while (true) {
+			try {
+				Thread.sleep(10);
+			} catch (InterruptedException e) {
+				e.printStackTrace();
+			}
+		}
+	}
+	
+	// ##########################################################
+	// #						METHODS							#
+	// ##########################################################
+	void oscEvent(OscMessage message) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.